### PR TITLE
Adding jks support in tidb MR connector

### PR DESCRIPTION
**Pull request by ankita25**
Currently only pem files are supported. This code adds support to jks

### CODE REVIEW

**Comment by sunxiaoguang** [comment]
Hi @ankita25, thanks for your contribution. Could you please verify if binary compiled with upgraded Hadoop version works with previous Hadoop version. From the changes it looks like they might be binary compatible. If that's the case maybe we can simply change dependency to provided and you can use whatever version you would like in production.

**Comment by ankita25** [comment]
> Hi @ankita25, thanks for your contribution. Could you please verify if binary compiled with upgraded Hadoop version works with previous Hadoop version. From the changes it looks like they might be binary compatible. If that's the case maybe we can simply change dependency to provided and you can use whatever version you would like in production.

I didn't mean to merge the hadoop changes. Let me remove them .. Thank you

**Comment by sunxiaoguang** [comment]
You can click on details link to find out why CI failed. 
For example this link:https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tibigdata_ghpr_build/detail/tibigdata_ghpr_build/415/pipeline/

**Comment by ankita25** [comment]
Is the integration test progressing ?

**Comment by sunxiaoguang** [comment]
you can write a comment like this '/run-all-tests' to trigger all tests

File: mapreduce/mapreduce-base/src/main/java/io/tidb/bigdata/mapreduce/tidb/TiDBConfiguration.java [hunk 1]
```diff
@@ -91,6 +96,17 @@ public static void clusterTls(Configuration conf,
...
+                                   String keyPath, String keyPassword, String trustPath,
+                                   String trustPassword) {
+    conf.set(CLUSTER_TLS_ENABLE, "true");
+    conf.set(CLUSTER_JKS_ENABLE, "true");
+    conf.set(CLUSTER_JKS_KEY_PATH, keyPath);
+    conf.set(CLUSTER_JKS_KEY_PASSWORD, keyPassword);
+    conf.set(CLUSTER_JKS_TRUST_PATH, trustPath);
+    conf.set(CLUSTER_JKS_TRUST_PASSWORD, trustPassword);
+  }
+
```
Comment (sunxiaoguang) [inline mapreduce/mapreduce-base/src/main/java/io/tidb/bigdata/mapreduce/tidb/TiDBConfiguration.java:109]: There are two empty spaces in this line

**Review by sunxiaoguang — COMMENTED** [review]
I'm wondering is it possible to add test for this new feature?

**Comment by ankita25** [comment]
> I'm wondering is it possible to add test for this new feature?

hmm I didn't see many tests for map reduce connector

**Comment by ankita25** [comment]
> I'm wondering is it possible to add test for this new feature?

The tests passed this time.

**Comment by sunxiaoguang** [comment]
> 
Sorry that I didn't make it clear. I know that existing code doesn't have enough tests, but we do hope it can get better over time. Therefore I asked if it is possible to add test to new code. If this feature can only be verified with integration test, we have to at least make sure it's tested manually.

**Comment by ankita25** [comment]
> > 
> 
> Sorry that I didn't make it clear. I know that existing code doesn't have enough tests, but we do hope it can get better over time. Therefore I asked if it is possible to add test to new code. If this feature can only be verified with integration test, we have to at least make sure it's tested manually.

yeah I did verify it manually
`

21/12/10 23:47:38 INFO mapreduce.Job: Job job_1639179006043_0006 running in uber mode : false
--
21/12/10 23:47:38 INFO mapreduce.Job:  map 0% reduce 0%
21/12/10 23:47:45 INFO mapreduce.Job:  map 100% reduce 0%
21/12/10 23:47:46 INFO mapreduce.Job: Job job_1639179006043_0006 completed successfully
21/12/10 23:47:46 INFO mapreduce.Job: Counters: 30
File System Counters
FILE: Number of bytes read=0
FILE: Number of bytes written=206810
FILE: Number of read operations=0
FILE: Number of large read operations=0
FILE: Number of write operations=0
HDFS: Number of bytes read=162
HDFS: Number of bytes written=0
HDFS: Number of read operations=1
HDFS: Number of large read operations=0
HDFS: Number of write operations=0
Job Counters
Launched map tasks=1
Other local map tasks=1
Total time spent by all maps in occupied slots (ms)=4836
Total time spent by all reduces in occupied slots (ms)=0
Total time spent by all map tasks (ms)=4836
Total vcore-milliseconds taken by all map tasks=4836
Total megabyte-milliseconds taken by all map tasks=4952064
Map-Reduce Framework
Map input records=1
Map output records=0
Input split bytes=162
Spilled Records=0
Failed Shuffles=0
Merged Map outputs=0
GC time elapsed (ms)=169
CPU time spent (ms)=9940
Physical memory (bytes) snapshot=306319360
Virtual memory (bytes) snapshot=2078961664
Total committed heap usage (bytes)=187695104
File Input Format Counters
Bytes Read=0
File Output Format Counters
Bytes Written=0
`
Test command - 
JAVA_HOME="/usr/lib/jvm/openjdk8" CLASSPATH=/usr/share/java/mysql-connector-java-8.0.25.jar /mnt/awagh/hadoop-2.10.0/bin/hadoop jar ~/mapreduce-tidb-connector-base-0.0.7-SNAPSHOT-PINTEREST.jar io.tidb.bigdata.mapreduce.tidb.example.TiDBMapreduceDemo  -du "jdbc:mysql://10.3.0.167:4000/test" -clusterTls  -clusterUseJks -clusterJksKeyPath <PATH> -clusterJksKeyPassword <PASSWORD> -clusterJksTrustPath <PATH> -clusterJksTrustPassword <PASSWORD>  -u  root  -p  ""  -dn test  -t test_table

**Comment by ankita25** [comment]
Any more concerns?

File: mapreduce/mapreduce-base/src/main/java/io/tidb/bigdata/mapreduce/tidb/TiDBConfiguration.java [hunk 2]
```diff
@@ -46,6 +46,11 @@
   public static final String CLUSTER_TLS_CA = "tidb.cluster-tls-ca";
   public static final String CLUSTER_TLS_KEY = "tidb.cluster-tls-key";
   public static final String CLUSTER_TLS_CERT = "tidb.cluster-tls-cert";
+  public static final String CLUSTER_JKS_ENABLE = "tidb.cluster-jks-enable";
+  public static final String CLUSTER_JKS_KEY_PATH = "tidb.cluster-jks-key-path";
+  public static final String CLUSTER_JKS_KEY_PASSWORD = "tidb.cluster-jks-key-password";
+  public static final String CLUSTER_JKS_TRUST_PATH = "tidb.jks.trust_path";
```
Comment (sunxiaoguang) [inline mapreduce/mapreduce-base/src/main/java/io/tidb/bigdata/mapreduce/tidb/TiDBConfiguration.java:52]: All configuration items uses dash however `trust_path` uses underscore instead. We should make it consistent.

**Review by sunxiaoguang — CHANGES_REQUESTED** [review]
Rest LGTM

**Comment by sunxiaoguang** [comment]
> 

Cool, thanks for your contribution. I left a comment about configuration key naming convention. Please take a look and that is the only comment I have.

**Comment by sunxiaoguang** [comment]
> Any more concerns?

Sorry, was busy training yesterday and didn't have chance to look at this PR.

**Comment by ankita25** [comment]
> Cool, thanks for your contribution. I left a comment about configuration key naming convention. Please take a look and that is the only comment I have.

updated. thanks

**Review by sunxiaoguang — APPROVED** [review]

**Pull request merged**